app/routes/goal_routes.py

```diff
@@ -0,0 +1,59 @@
+from flask import Blueprint, make_response, request
+from app.models.goal import Goal
+from app.models.task import Task
+from app.routes.helpers import validate_model, create_item, get_all_items, get_one_item, update_item, delete_item
+from app import db
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    return create_item(Goal)
+
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    return get_all_items(Goal)
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    return get_one_item(Goal, goal_id)
+    
+
+@goals_bp.route("<goal_id>", methods=["PUT"])
+def update_task(goal_id):
+    return update_item(Goal, goal_id)
+    
+
+@goals_bp.route("<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    return delete_item(Goal, goal_id)
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=['POST'])
+def create_goal_with_tasks(goal_id):
+
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    task_ids = request_body.get("task_ids")
```

Comment:
I like that you used .get() to get the task_ids. Consider passing a default value that is an empty list, like: `task_ids = request_body.get("task_ids", [])`

As is, if the key "task_ids" is not in request_body then the variable `task_ids` will reference None. Then when you try to loop over None, you'll run into another unhandled exception (Nonetype is not iterable or something like that)